src/main/java/com/example/demo/service/UrlExtractor.java

```diff
@@ -0,0 +1,21 @@
+package com.example.demo.service;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class UrlExtractor {
+
+    public static String extractURLFromJSON(String message) {
+
+
+        // Proba bezposredniego wyciagniecia URL z pola "message"
+        Pattern pattern = Pattern.compile("(https?://\\S+)");
+        Matcher matcher = pattern.matcher(message);
+
+        if (matcher.find()) {
+            return matcher.group(1);
+        }
+
+        return null;
+    }
```

Comment:
Mam tutaj kilka spostrzeżeń: 
- Czy ten regex wystarczy by znaleźć wszystkie linki w wiadomości?
- Co jeżeli mamy więcej niż 1 adres w wiadomości?
- Dlaczego metoda nazywa się `extractURLFromJSON` -> `JSON` 
- Co moglibyśmy zwrócić zamiast null'a by nie musieć później radzić sobie z nullami? Wiesz może co w Javie/Scali jest takiego, co ułatwia pracę z nullami? Podpowiem Ci, że zwracany typ w definicji tej metody lekko się zmieni :slightly_smiling_face:

Reply:
no to zaproponuje taką metodę:
public static Optional<List<String>> extractUrlsFromMessage(String message) {
        Pattern pattern = Pattern.compile("(https?://\\S+)");
        Matcher matcher = pattern.matcher(message);
        List<String> urls = new ArrayList<>();

        while (matcher.find()) {
            urls.add(matcher.group());
        }

        return urls.isEmpty() ? Optional.empty() : Optional.of(urls);
    }
    
    Teraz mamy listę url'i - jeszcze przechowuje te linki w Stringu, ale ustalaliśmy wyżej że typ URL by był pewnie bardziej optymalny. 
    Później tylko trzeba by te strony sprawdzać osobno, bo zgodnie z dokumentacją API przyjmuje pojedynczy argument a nie listę argumentów. 
   
   Ten zwracany null może prowadzić do NullPointerException no i chyba moglibyśmy tego uniknąć używając @Optional i w taki sposób obsłużyć nulla :)